src/BloomExe/ImageProcessing/RuntimeImageProcessor.cs

```diff
@@ -0,0 +1,201 @@
+	{
+		private readonly BookRenamedEvent _bookRenamedEvent;
+		public int TargetDimension = 500;
+		private Dictionary<string, string> _originalPathToProcessedVersionPath;
+		private string _cacheFolder;
+
+		private ImageAttributes _transparentImageAttributes;
+
+		public RuntimeImageProcessor(BookRenamedEvent bookRenamedEvent)
+		{
+			_bookRenamedEvent = bookRenamedEvent;
+			_originalPathToProcessedVersionPath = new Dictionary<string, string>();
+			_cacheFolder = Path.Combine(Path.GetTempPath(), "Bloom");
+			_bookRenamedEvent.Subscribe(OnBookRenamed);
+			_transparentImageAttributes = new ImageAttributes();
+			_transparentImageAttributes.SetColorKey(Color.FromArgb(253, 253, 253), Color.White);
+		}
+
+		private void OnBookRenamed(KeyValuePair<string, string> fromPathAndToPath)
+		{
+			//Note, we don't pay attention to what the change was, we just purge the whole cache
+
+			TryToDeleteCachedImages();
+			_originalPathToProcessedVersionPath = new Dictionary<string, string>();
+		}
+
+		public void Dispose()
+		{
+			if (_originalPathToProcessedVersionPath == null)
+				return;
+
+			TryToDeleteCachedImages();
+			_originalPathToProcessedVersionPath = null;
+
+			//NB: this turns out to be dangerous. Without it, we still delete all we can, leave some files around
+			//each time, and then deleting them on the next run
+			//			_cacheFolder.Dispose();
+
+			GC.SuppressFinalize(this);
+		}
+
+		private void TryToDeleteCachedImages()
+		{
+			lock (this)
+			{
+				foreach(var path in _originalPathToProcessedVersionPath.Values)
+				{
+					try
+					{
+						if (File.Exists(path))
+						{
+							File.Delete(path);
+							Debug.WriteLine("RuntimeImageProcessor Successfully deleted: " + path);
+						}
+					}
+					catch (Exception e)
+					{
+						Debug.WriteLine("RuntimeImageProcessor Dispose(): " + e.Message);
+					}
+				}
+				_originalPathToProcessedVersionPath.Clear();
+			}
+		}
+
+		public string GetPathToResizedImage(string originalPath)
+		{
+			//don't mess with Bloom UI images
+			if (new[] {"/img/", "placeHolder", "Button"}.Any(s => originalPath.Contains(s)))
+				return originalPath;
+
+			lock (this)
+			{
+				string pathToProcessedVersion;
+				if (_originalPathToProcessedVersionPath.TryGetValue(originalPath, out pathToProcessedVersion))
+				{
+					if (File.Exists(pathToProcessedVersion) &&
+					    new FileInfo(originalPath).LastWriteTimeUtc <= new FileInfo(pathToProcessedVersion).LastWriteTimeUtc)
+					{
+						return pathToProcessedVersion;
+					}
+					else
+					{
+						_originalPathToProcessedVersionPath.Remove(originalPath);
+					}
+				}
+
+				using (var originalImage = PalasoImage.FromFile(originalPath))
+				{
+					//if it's a jpeg, we don't resize, we don't mess with transparency, nothing. These things
+					//are scary in .net. Just send the original back and wash our hands of it.
+					if (ImageUtils.AppearsToBeJpeg(originalImage))
+					{
+						return originalImage.OriginalFilePath;
```

Comment:
Seems it would be shorter and perhaps marginally clearer to just return originalPath